src/ring_theory/graded_algebra/basic.lean

```diff
@@ -166,6 +166,70 @@ lemma graded_algebra.mem_support_iff [decidable_eq A] (r : A) (i : ι) :
   i ∈ (decompose 𝒜 r).support ↔ graded_algebra.proj 𝒜 i r ≠ 0 :=
 dfinsupp.mem_support_iff.trans submodule.coe_eq_zero.not.symm
 
+lemma graded_algebra.proj_homogeneous_mul (add_left_cancel : ∀ {i j k : ι}, i + j = i + k → j = k)
```

Comment:
This feels like it should be stated in terms of `graded_ring.proj`, since it doesn't require submodules to be true.

Also, I think the `add_left_cancel` assumption would be better handled by assuming `add_left_cancel_monoid`